integrations/java/openhouse-java-runtime/src/main/java/com/linkedin/openhouse/javaclient/exception/WebClientResponseWithMessageException.java

```diff
@@ -8,21 +8,36 @@
  *
  * <p>In this class we override the {@link #getMessage()} behavior, to append server response if it
  * exists. It doesn't change status code (ex: forbidden) or status int (ex. 403)
+ *
+ * <p>An exception thrown in Openhouse clients to indicate an error in a tables API request, acting
+ * as a wrapper around WebClientResponseException.
  */
-public class WebClientResponseWithMessageException extends WebClientResponseException {
-  public WebClientResponseWithMessageException(WebClientResponseException e) {
-    super(
+public class WebClientResponseWithMessageException extends WebClientWithMessageException {
+  private final WebClientResponseException responseException;
+
+  /**
+   * We want the response exception to go through the legacy constructor in {@link
+   * WebClientResponseException} because the HttpRequest value is explicitly set to null in order to
+   * have the message rewritten to not expose unnecessary data.
+   */
+  public WebClientResponseException createWebClientResponseException(WebClientResponseException e) {
```

Comment:
Should this be `public`? By keeping it `public`, we still end up exposing the shaded `WebClientResponseException`.

Reply:
we can make this `private`, since it's only being called by the constructor